runtime/src/main/java/org/corfudb/protocols/service/CorfuProtocolMessage.java

```diff
@@ -0,0 +1,145 @@
+     */
+    public static HeaderMsg getHeaderMsg(long requestId, PriorityLevel priority, long epoch, UuidMsg clusterId,
+                                         UuidMsg clientId, boolean ignoreClusterId, boolean ignoreEpoch) {
+        return HeaderMsg.newBuilder()
+                .setVersion(CURRENT_VERSION)
+                .setRequestId(requestId)
+                .setPriority(priority)
+                .setEpoch(epoch)
+                .setClusterId(clusterId)
+                .setClientId(clientId)
+                .setIgnoreClusterId(ignoreClusterId)
+                .setIgnoreEpoch(ignoreEpoch)
+                .build();
+    }
+
+    /**
+     * Returns a header containing information common to all service RPCs.
+     *
+     * @param requestId         the request id, generated by the client
+     * @param priority          the priority of the message
+     * @param epoch             the epoch of the message
+     * @param clusterId         the cluster id
+     * @param clientId          the clients own id
+     * @param ignoreClusterId   indicates if the message is clusterId aware
+     * @param ignoreEpoch       indicates if the message is epoch aware
+     */
+    public static HeaderMsg getHeaderMsg(long requestId, PriorityLevel priority, long epoch, UUID clusterId,
+                                         UUID clientId, boolean ignoreClusterId, boolean ignoreEpoch) {
+        return getHeaderMsg(requestId, priority, epoch, getUuidMsg(clusterId),
+                getUuidMsg(clientId), ignoreClusterId, ignoreEpoch);
+    }
+
+    /**
+     * Returns a header containing information common to all service RPCs.
+     * Used by the server as a convenient way to copy reusable field values.
+     *
+     * @param header            the original request header
+     * @param ignoreClusterId   indicates if the message is clusterId aware
+     * @param ignoreEpoch       indicates if the message is epoch aware
+     */
+    public static HeaderMsg getHeaderMsg(HeaderMsg header, boolean ignoreClusterId, boolean ignoreEpoch) {
+        return getHeaderMsg(header.getRequestId(),
+                header.getPriority(),
+                header.getEpoch(),
+                header.getClusterId(),
+                header.getClientId(),
+                ignoreClusterId,
+                ignoreEpoch);
+    }
+
+    /**
+     * Returns a header containing information common to all service RPCs.
+     * Use by the server to increase the priority of certain requests.
+     *
+     * @param header   the original request header
+     */
+    public static HeaderMsg getHighPriorityHeaderMsg(HeaderMsg header) {
+        return getHeaderMsg(header.getRequestId(),
+                PriorityLevel.HIGH,
+                header.getEpoch(),
+                header.getClusterId(),
+                header.getClientId(),
+                header.getIgnoreClusterId(),
+                header.getIgnoreEpoch());
```

Comment:
Can't you do `HeaderMsg.newBuilder().mergeFrom(header).setPriority(PriorityLevel.HIGH);`?